File: OneCard.py
class OneCard: 
    owner: str = ""
    balance: float = 0.0

    def __init__(self, name, starting_balance) -> None:
        self.owner = name
        self.balance = starting_balance

    def CanAfford(self, cost) -> bool:
        if (cost <= self.balance):
            return True
        else: #cost > self.balance
            return False

    def deposit(self, amt_deposit) -> None: 
        self.balance += amt_deposit

    def spend(self, amt_deposit) -> None: 
        if self.CanAfford(amt_deposit):
            self.balance -= amt_deposit 
        else:
            print("insufficient funds")

def main():
    my_card = OneCard("Abbey", 10.0)
    print(my_card.owner, "has", my_card.balance, "on card")
    print(my_card.CanAfford(200))
    my_card.deposit(2000)
    print(my_card.owner, "has", my_card.balance, "on card")
    print(my_card.CanAfford(200))


main()

File: onecard.java
/**
 * Example One Card Class 
 */
public class OneCard {
    /** store card owners name */
    private String owner;
    /** store card balance of owner */
    private double balance;

/** 
 * Constructor Summary 
*/
public OneCard(String person, double money){
    this.owner = person;
    this.balance = money;
}

public void deposit(double amount){
    this.balance += amount;
}

public boolean canAfford(double amount){
    return this.balance > amount;
}

public void spend(double amount){
    if(!this.canAfford(amount)){ 
        System.out.println("you can not afford this");
    }
    else{ 
        this.balance -= amount; 
    }
    
}

public static void main(String[] args){
    OneCard myOneCard = new OneCard("alex", 472.35);
    //System.out.println(myOneCard);
    OneCard myFriendCard = new OneCard("Emily", 10.02);
    System.out.println(myFriendCard.balance);
    System.out.println(myOneCard.canAfford(50));
    System.out.println(myFriendCard.canAfford(50));
}

}



/**
 * checks balance available on this card
 * @param 
 * 
 */